localapi/localapi_transport_http.go

```diff
@@ -47,7 +55,28 @@ func setHeaders(responseWriter *http.ResponseWriter) {
 	(*responseWriter).Header().Set("content-type", "application/json")
 }
 
+
+func checkAuth(w http.ResponseWriter, r *http.Request) bool {
+	s := strings.SplitN(r.Header.Get("Authorization"), " ", 2)
+	if len(s) != 2 { return false }
+
+	b, err := base64.StdEncoding.DecodeString(s[1])
+	if err != nil { return false }
+
+	pair := strings.SplitN(string(b), ":", 2)
+	if len(pair) != 2 { return false }
+
+	return pair[0] == "Disgo" && pair[1] == "Dance"
+}
+
+
 func (this *LocalAPIService) tranferHandler(responseWriter http.ResponseWriter, request *http.Request) {
+	if !checkAuth(responseWriter, request) {
+		responseWriter.Header().Set("WWW-Authenticate", `Dis realm="Dispatch Local"`)
```

Comment:
Please see the spec on this header:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate
We must follow that. "Dis realm" is not a thing.

Reply:
sorry dave, my fault for lack of knowledge